### ISSUE TITLE
Remove create() and delete() methods from reaction object

### ISSUE THREAD

**Issue by JDGrimes** [WordPoints/hooks-api#35]
The reaction class currently includes `create()` and `delete()` methods. In this issue I'd like to explore the possibility that we should remove them.

The biggest problem is with the `create()` method. To understand it, first some background.
## Background

One of the big things that I wanted this new API to accomplish is to help bring some more sanity to the way that the reactions are stored. Early on, I decided that we should make it possible to store the reactions in any way we wanted to, instead of confining them to the options table, as was previously the case. This led to the introduction of a reaction storage interface, with methods to create, update, and retrieve the reactions. One storage class could then store reactions in the options table, while another could store them in the posts table (or outside of the DB entirely, if it wanted to).

Each reactor is then given the option of which storage class (and thus storage type) it wants to use for reactions associated with it.

The storage class did the work of creating updating and retrieving the reactions as single units. However, something more was needed for the API to be truly storage agnostic. We also needed a way to CRUD just specific bits of "metadata" for each reaction. The solution to this was to represent the reactions with special objects in the API. Each of these objects would implement a common interface. So, just like the reaction storage classes, we could manipulate the reactions without having to worry about how they are actually stored.
## Current State

This can be mildly confusing, however, because we now must have two classes for each different way that we want to store reactions: the reaction storage class, and the reaction object class. This necessarily requires that there be a large amount of coupling between the two classes for each implementation.

Because of this, what parts of the implementation belonged in which class was not always clear in my mind. This led to the basic bootstrap for updating and creating the reactions (like validation, etc.) being placed in the storage class, while the actual creation of the reaction was placed in the reaction class—in the `create()` method.
## Issues

This, in turn, makes things more confusing. Because now we have this create method that does not perform any validation, or indeed even check if the reaction already exists. And because this method exists alongside `add_meta()` et. al. it is easy to wonder whether a flow like this is possible (it isn't):

``` php
$reaction = new WordPoints_Hook_Reaction_Options();
$reaction->add_meta( 'key', 'value' );
$reaction->add_meta( 'another', 'value' );
$reaction->create();
```

In other words, the reaction class now has two unconnected flows: the `create()` method, and the `*_meta()` methods.
## Solution

The best way forward, I think, is to stick rigidly to the basic definition I gave for each class above:
- The storage class should CRUD reactions,
- The reaction class should CRUD reaction meta.

Clean and simple.

**Comment by JDGrimes** [comment]
I could also note that there isn't really any precedent for the `create()` method anyway, as far as `WP_Post` is concerned, which is sort of what the class is based off of originally (via `WordPoints_Rank`, IIRC).

**Comment by JDGrimes** [comment]
We also have an `exists()` method, which is sort of anomalous after the `create()` and `delete()` methods have been removed. Though `WP_Post` does in fact have an `exists()` method, so there is some precedent for that. But I think that it would make more sense to have an `reaction_exists()` method on the storage class, because the storage class already knows how to retrieve the reactions, while the object class doesn't otherwise have to (anymore). We could still offer the `exists()` method on the reaction class, however, it really wouldn't be necessary, because our flow is such that it would only be used by the storage class anyway.
